### ISSUE TITLE
[Feature-request] Migrate admin template to Angular 2

### ISSUE THREAD

**Issue by codespicelabs** [pyrocms/pyrocms#3933]
Hi there!

Pyro is very interesting CMF for Laravel folks in my humble opinion, but i have a little different point of view how the modern admin UI [should look like for now](https://github.com/jhipster/generator-jhipster/milestone/79) ;)

1. What about Angular 2 for admin theme and more closely to SPA instead of full page reloading after some slight changes have been done in the admin area ?

2. Please provide some tips and "best practices" how to integrate a separate module (addon in Pyro terms) been written in Angular 2 into PyroCMS admin UI that the module would be able to communicate with the rest of admin's JS parts (Pyro panel's button actions etc). The modules (packages) were written for Laravel surely with custom Admin theme made with BS4 and Angular 2.
I couldn't find anything about admin UI customization and other relative twiks in your docs.
Kind regards.

**Comment by RyanThompson** [comment]
Hey thanks for your input!

There are a few really good reasons I approached the UI the way I did.

1.) Angular instantly adds more learning curve and by principle more complexity. 

PyroCMS is mega powerful but things can be accomplished very quickly and cleanly. A lot of this has to do with the dumb-easy reload approach and structured default application patterns. 

In order for people to get in and start building as fast as possible. This is the way it has to be.

2.) It's new!

I love angular apps - and I want Pyro to mature into more angular sexiness. But the system is just too new. I for one am not too much of a guru with angular. I am learning and developing ideas but the same rapid deployment principles that make Pyro SO easy to build on now - need to be kept in mind and developed for as we start adding Angular resources. I just don't know what that looks like yet ^_^ All ears if you have ideas though.

3.) The control panel is not for developers. It's for users. 

While developers need to be able to work proficiently in the control panel the UI _must_ accomodate for the lowest common denominator. The everyday users! The people who will be managing their application or website with Pyro. 

That said - I have some awesome ideas for how to push the UI forward from a design perspective and keep everyone happy I think. But again - this takes time and it's something that needs to be thought out thoroughly because it will be the last REAL big UI change for quite some time (if any is decided upon).

----

Regarding tip and docs - they are CONSTANTLY growing. I am finishing up a videos module that will be launching this week with video tutorials as well writing developer docs hopefully tonight / tomorrow. They'll be a bear I am sure. But the docs are always being added to - feel free to post suggestions that are more concise. The forum is also a good place to discuss / request documentation stuff as well as general support. And our slack team ^_^

I hope this sheds some light on why things are and where we're going! 

Cheers for now,

Ryan
